py-polars/tests/unit/test_interop.py

```diff
@@ -476,6 +476,47 @@ def test_from_arrow() -> None:
     assert df.rows() == [(1, 4), (2, 5), (3, 6)]  # type: ignore[union-attr]
     assert df.schema == {"a": pl.UInt32, "b": pl.UInt64}  # type: ignore[union-attr]
 
+    # empty string column name
+    df = pl.select(pl.Series("", [1], pl.Int8), pl.Series("a", [1], pl.Int8))
+    dfs = (df.select(""), df)
+
+    for df in dfs:
+        assert pl.DataFrame(df.to_arrow()).schema == df.schema
+        assert pl.DataFrame(df.head(0).to_arrow()).schema == df.schema
+
+
+def test_from_arrow_duplicate_names() -> None:
+    a1 = pa.array([1])
+    a2 = pa.array([2])
+    tbl_base = pa.Table.from_arrays([a1, a2], names=["x", "x"])
+
+    for tbl, err_match in (
+        (
+            tbl_base.rename_columns(["", ""]),
+            r"^Arrow table contained duplicate column ''.*",
+        ),
+        (
+            tbl_base.rename_columns(["a", "a"]),
+            r"^Arrow table contained duplicate column 'a'.*",
+        ),
+    ):
+        with pytest.raises(ValueError, match=err_match):
+            pl.DataFrame(tbl)
+
+        assert pl.DataFrame(tbl, schema={"": pl.Int8, "a": pl.Int8}).to_dicts() == [
+            {
+                "": 1,
+                "a": 2,
+            }
+        ]
+
+        assert pl.DataFrame(tbl, schema=["", ""]).to_dicts() == [
```

Comment:
This should also raise - it's a duplicate column name. We should not have special logic for empty column names.

Reply:
Ahh - that one actually affects any sort of construction using the schema argument (not just arrow), so I held off on it. But I'll give it a try - perhaps it won't break too many things.

Reply:
Yeah that is what makes this one tricky.. let me know what you run into!

Reply:
I've updated the PR to raise for this, and the issue description as well. As far as the existing test cases go, the only complication was the behavior of `DataFrame` construction from lists of `Series`, which is preserved as is for now (see in the updated issue description).

Reply:
Looks like I missed a spot - for the `scan_csv` and `read_csv` functions - they currently also don't raise https://github.com/pola-rs/polars/issues/11707. Will put this back in draft and update those as well (as well as any others).